### PR TITLE
Add `#ifdef` and `#ifndef` as `brackets`

### LINKED ISSUE #160744: #ifdef/endif colorization matching doesn't work correctly with C/C++
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: Version: 1.72.0-insider (user setup)
Commit: 835ace5796cec0ed19a7eec119b26b57220b0f1a
Date: 2022-09-12T05:16:16.290Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: Yes
It doesn't repro with non-Insiders.

Steps to Reproduce:

1. Use a .c or .cpp file with
```cpp
#ifdef FOO
#endif
```

Bug: The matching code doesn't handle this case (and #ifndef, #else, #elif, #elifdef, #elifndef):
image

The C/C++ extension is not enabled.

### PR DESCRIPTION
**Pull request by RedCMD**
Fixes https://github.com/microsoft/vscode/issues/160744
extension of https://github.com/microsoft/vscode/pull/159997 (https://github.com/microsoft/vscode/issues/159586)

image

Cannot add `#else`, `#elif`, `#elifdef` or `#elifndef` as `brackets`
as Vscode does not support 'middle'/'inside' `brackets`
however, it does not cause any issues

### CODE REVIEW

**Comment by sean-mcmanus** [comment]
@hediet This is important to get checked in ASAP -- otherwise the C/C++ experience is regressed with VS Code Insiders, i.e. red `#endif` appear all over the place with code bases that have heavy `#ifdef/#ifndef` usage.

File: extensions/cpp/language-configuration.json [hunk 1]
```diff
@@ -7,7 +7,9 @@
 		["{", "}"],
 		["[", "]"],
 		["(", ")"],
-		["#if","#endif"],
+		["#ifndef", "#endif"],
+		["#ifdef", "#endif"],
+		["#if", "#endif"],
```
Comment (sean-mcmanus) [inline extensions/cpp/language-configuration.json:12]: Should the trailing `,` be removed in order to comply with standard json parsing rules? A user was complaining about that at https://github.com/microsoft/vscode/pull/159997#issuecomment-1250634995

**Review by sean-mcmanus — APPROVED** [review]

**Comment by hediet** [comment]
Closing, as we revert the `#...` brackets for cpp in #161589.

**Comment by sean-mcmanus** [comment]
@hediet Okay, that seems fine. What were the problems though? I was only aware of the problem with `#endif` not matching with `#ifdef` and `#ifndef` which got fixed by this PR.

**Comment by hediet** [comment]
@sean-mcmanus you reported the problem here: https://github.com/microsoft/vscode/issues/159586#issuecomment-1248899801


This one is not fixable.

**Pull request closed without merging**